### PR TITLE
hw 5.1 solution

### PR DESCRIPTION
**Pull request by VBortnyk**

### CODE REVIEW

File: src/main/java/core/basesyntax/Engine.java [hunk 1]
```diff
@@ -1,5 +1,66 @@
 package core.basesyntax;
 
-public class Engine {
+public class Engine implements Cloneable {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Engine.java:3]: for what do we need to implement Cloneable?

File: src/main/java/core/basesyntax/Car.java [hunk 2]
```diff
@@ -1,5 +1,64 @@
...
+        return colour;
+    }
+
+    public void setColour(String colour) {
+        this.colour = colour;
+    }
+
+    @Override
+    public Object clone() throws CloneNotSupportedException {
+        return super.clone();
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Car.java:62]: don't use super.clone() it's not a deep cloning

File: src/main/java/core/basesyntax/Car.java [hunk 3]
```diff
@@ -1,5 +1,66 @@
...
+        return colour;
+    }
+
+    public void setColour(String colour) {
+        this.colour = colour;
+    }
+
+    @Override
+    public Object clone() throws CloneNotSupportedException {
+        return new Car(brand, model, price, weight, colour, engine);
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Car.java:63]: can we remove throws and change what our method returns?

File: src/main/java/core/basesyntax/Car.java [hunk 4]
```diff
@@ -1,5 +1,66 @@
...
+        return colour;
+    }
+
+    public void setColour(String colour) {
+        this.colour = colour;
+    }
+
+    @Override
+    public Object clone() {
+        return new Car(brand, model, price, weight, colour, engine);
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Car.java:63]: that would be a new Car but with the old engine

Comment (VBortnyk) [reply src/main/java/core/basesyntax/Car.java:63]: done

Comment (dmytro-hryhoruk) [reply src/main/java/core/basesyntax/Car.java:63]: where

File: src/main/java/core/basesyntax/Engine.java [hunk 5]
```diff
@@ -1,5 +1,66 @@
...
+    public int getPrice() {
+        return price;
+    }
+
+    public void setPrice(int price) {
+        this.price = price;
+    }
+
+    @Override
+    protected Object clone() throws CloneNotSupportedException {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Engine.java:63]: can we remove throws?

File: src/main/java/core/basesyntax/Car.java (hunk 4 above)
Comment (dmytro-hryhoruk) [reply src/main/java/core/basesyntax/Car.java]: ?

File: src/main/java/core/basesyntax/Car.java [hunk 6]
```diff
@@ -1,5 +1,66 @@
...
+    public String getColour() {
+        return colour;
+    }
+
+    public void setColour(String colour) {
+        this.colour = colour;
+    }
+
+    @Override
+    public Object clone() {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Car.java:62]: let's return Car

File: src/main/java/core/basesyntax/Car.java [hunk 7]
```diff
@@ -1,5 +1,66 @@
...
+        return new Car(brand, model, price, weight, colour, new Engine());
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Car.java:63]: the engine of original Car had some field(and values in it), but this Engine would be empty

Comment (dmytro-hryhoruk) [reply src/main/java/core/basesyntax/Car.java]: you should use engine.clone() here

Comment (dmytro-hryhoruk) [reply src/main/java/core/basesyntax/Car.java]: make sure your solution passes all tests

Comment (VBortnyk) [reply src/main/java/core/basesyntax/Car.java]: have a look please

File: src/main/java/core/basesyntax/Car.java [hunk 8]
```diff
@@ -1,5 +1,66 @@
...
+        return colour;
+    }
+
+    public void setColour(String colour) {
+        this.colour = colour;
+    }
+
+    @Override
+    public Object clone() {
+        return new Car(brand, model, price, weight, colour, (Engine)engine.clone());
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Car.java:63]: we don't need to cast it

File: src/main/java/core/basesyntax/Car.java (hunk 6 above)
Comment (dmytro-hryhoruk) [reply src/main/java/core/basesyntax/Car.java:62]: make sure you read all the comments

**Pull request closed without merging**